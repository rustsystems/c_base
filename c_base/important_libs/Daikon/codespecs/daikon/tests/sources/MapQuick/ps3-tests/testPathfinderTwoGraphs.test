#Test that there's no aliasing between two graphs for PathFinder
CreateNode n1 2
CreateNode n2 20
CreateNode n3 4
CreateNode n4 4
CreateNode n5 12
CreateNode n6 1
CreateNode n7 3
CreateNode n8 9
CreateNode n9 5
CreateNode n10 3
CreateNode n11 11
#set up 1st graph
CreateGraph A
AddNode A n1
AddNode A n2
AddNode A n3
AddNode A n4
AddNode A n5
AddNode A n6
AddNode A n7
AddNode A n8
AddNode A n9
AddNode A n10
AddNode A n11
AddEdge A n1 n2
AddEdge A n1 n5
AddEdge A n1 n8
AddEdge A n2 n3
AddEdge A n2 n7
AddEdge A n3 n4
AddEdge A n4 n3
AddEdge A n4 n7
AddEdge A n4 n11
AddEdge A n5 n1
AddEdge A n5 n2
AddEdge A n5 n6
AddEdge A n6 n9
AddEdge A n6 n10
AddEdge A n7 n11
AddEdge A n8 n5
AddEdge A n8 n9
AddEdge A n9 n5
AddEdge A n9 n6
AddEdge A n9 n10
AddEdge A n10 n7
AddEdge A n11 n10
#add more nodes & edges in a disjoint set
CreateNode n20 1
CreateNode n21 2
CreateNode n22 3
AddNode A n20
AddNode A n21
AddNode A n22
AddEdge A n20 n21
AddEdge A n20 n22
AddEdge A n21 n22
AddEdge A n21 n20
AddEdge A n22 n20
AddEdge A n22 n21
#create a second graph and add some of the same nodes to it
CreateGraph B
AddNode B n1
AddNode B n2
AddNode B n3
AddNode B n20
AddNode B n21
AddEdge B n1 n20
AddEdge B n20 n21
AddEdge B n21 n2
AddEdge B n2 n1

#do some pathfinding that would work in 1st graph, but not in 2nd
FindPath A n1 -> n3

#do some pathfinding that would work in 2nd, but not in 1st
FindPath B n1 -> n21
