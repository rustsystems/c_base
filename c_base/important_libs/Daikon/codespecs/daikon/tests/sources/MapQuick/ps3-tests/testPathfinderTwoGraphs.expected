#Test that there's no aliasing between two graphs for PathFinder
created node n1 with cost 2
created node n2 with cost 20
created node n3 with cost 4
created node n4 with cost 4
created node n5 with cost 12
created node n6 with cost 1
created node n7 with cost 3
created node n8 with cost 9
created node n9 with cost 5
created node n10 with cost 3
created node n11 with cost 11
#set up 1st graph
created graph A
added node n1 to A
added node n2 to A
added node n3 to A
added node n4 to A
added node n5 to A
added node n6 to A
added node n7 to A
added node n8 to A
added node n9 to A
added node n10 to A
added node n11 to A
added edge from n1 to n2 in A
added edge from n1 to n5 in A
added edge from n1 to n8 in A
added edge from n2 to n3 in A
added edge from n2 to n7 in A
added edge from n3 to n4 in A
added edge from n4 to n3 in A
added edge from n4 to n7 in A
added edge from n4 to n11 in A
added edge from n5 to n1 in A
added edge from n5 to n2 in A
added edge from n5 to n6 in A
added edge from n6 to n9 in A
added edge from n6 to n10 in A
added edge from n7 to n11 in A
added edge from n8 to n5 in A
added edge from n8 to n9 in A
added edge from n9 to n5 in A
added edge from n9 to n6 in A
added edge from n9 to n10 in A
added edge from n10 to n7 in A
added edge from n11 to n10 in A
#add more nodes & edges in a disjoint set
created node n20 with cost 1
created node n21 with cost 2
created node n22 with cost 3
added node n20 to A
added node n21 to A
added node n22 to A
added edge from n20 to n21 in A
added edge from n20 to n22 in A
added edge from n21 to n22 in A
added edge from n21 to n20 in A
added edge from n22 to n20 in A
added edge from n22 to n21 in A
#create a second graph and add some of the same nodes to it
created graph B
added node n1 to B
added node n2 to B
added node n3 to B
added node n20 to B
added node n21 to B
added edge from n1 to n20 in B
added edge from n20 to n21 in B
added edge from n21 to n2 in B
added edge from n2 to n1 in B

#do some pathfinding that would work in 1st graph, but not in 2nd
shortest path in A: n1 n2 n3

#do some pathfinding that would work in 2nd, but not in 1st
shortest path in B: n1 n20 n21
